connectrum/svr_info.py

```diff
@@ -158,14 +159,17 @@ def from_irc(self, irc_nickname=None, irc_password=None):
         # merge by nick name
         self.update(results)
 
-    def add_single(self, hostname, ports, nickname=None, **kws):
+    def add_single(self, hostname, ports=None, nickname=None, **kws):
         '''
             Explicitly add a single entry.
             Hostname is a FQDN and ports is either a single int (assumed to be TCP port)
             or Electrum protocol/port number specification with spaces in between.
         '''
-        nickname = nickname or hostname
+        if isinstance(hostname, ServerInfo):
+            self[str(hostname)] = hostname
```

Comment:
It's already called in [`ServerInfo.__str__`](https://github.com/iamale/connectrum/blob/76e76a464250e8ba092da606573b32cdb431e8c2/connectrum/svr_info.py#L118) (and `hostname` here is `ServerInfo` actually), so no seed to.

Maybe we can rename it to `hostname_or_svr_info`, but it's too verbose in my opinion..?